src/AzureClient/Mocks/MockCloudJob.cs

```diff
@@ -7,9 +7,89 @@
 using Azure.Quantum.Jobs.Models;
 using System;
 using System.IO;
+using System.Collections.Generic;
 
 namespace Microsoft.Quantum.IQSharp.AzureClient
 {
+    public class MockCostEstimate
```

Comment:
It definitely makes me a bit uneasy how much of this uses reflection to bypass visibility modifiers. I understand how that's forced by the fact that the autogenerated classes in Azure.Quantum.Jobs are immutable and only have `internal` constructors, so I don't even have a better alternative to suggest, but it may be good to add some comments indicating why this is needed. Hopefully, we can file an upstream feature request to the code generator used as well?